src/nvim/os/fs.c

```diff
@@ -145,7 +145,19 @@ static bool is_executable_in_path(const char_u *name, char_u **abspath)
     return false;
   }
 
+#ifdef WIN32
+  const char *pathext = os_getenv("PATHEXT");
+  size_t buf_len;
+
+  if (pathext) {
+    buf_len = STRLEN(name) + STRLEN(path) + 2 + STRLEN(pathext);
+  } else {
+    pathext = ".com;.exe;.bat;.cmd";
+    buf_len = STRLEN(name) + STRLEN(path) + 2 + 4;
+  }
+#else
   size_t buf_len = STRLEN(name) + STRLEN(path) + 2;
+#endif
```

Comment:
Suggestion: eliminate the `else{}` and `#else`.

```c
  size_t buf_len = STRLEN(name) + STRLEN(path) + 2;
#ifdef WIN32
  const char *pathext = os_getenv("PATHEXT");
  if (!pathext) {
    pathext = ".com;.exe;.bat;.cmd";
  }

  buf_len += STRLEN(pathext);
#endif
```

The gain in readability is worth the small increase in `buf_len`, and searching the path is far more expensive...